src/window/window_wrapper/keyboard_manager.rs

```diff
@@ -166,11 +165,14 @@ impl KeyboardManager {
                                     .send(UiCommand::Keyboard(keybinding_string))
                                     .expect("Could not send keyboard ui command");
                             } else {
-                                let key_text = if self.alt && cfg!(target_os = "macos") {
-                                    key_event.text_with_all_modifiers()
-                                } else {
-                                    key_event.text
-                                };
+                                let is_dead_key = key_event.text_with_all_modifiers().is_some()
+                                    && key_event.text.is_none();
+                                let key_text =
+                                    if (self.alt || is_dead_key) && cfg!(target_os = "macos") {
+                                        key_event.text_with_all_modifiers()
```

Comment:
I thought of doing something like `let text_with_mods = key_event.text_with_all_modifiers()` at the start of the `else` and using that so `text_with_all_modifiers()` isn't called twice, but I'm not sure if that's necessary?

Reply:
yeah I'm pretty sure it doesn't matter too much. Im gonna leave it as its more concise this way